File: scratch/deploy_vps.py
import os
import sys
import paramiko

def main():
    hostname = '192.168.1.70'
    username = 'ed'
    password = '2494'
    remote_dir = 'projects/BelaFarma'
    local_env_path = r'f:\Documentos\Desenvolvimento\BelaFarma\.env'
    remote_env_path = f'{remote_dir}/.env'

    print("=== SSH/SFTP DEPLOYMENT TO BELAFARMA ===")
    
    # Initialize SSH client
    ssh = paramiko.SSHClient()
    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
    
    try:
        print(f"Connecting to {username}@{hostname}...")
        ssh.connect(hostname, username=username, password=password, timeout=15)
        print("Connected successfully!")
        
        # 1. SFTP Upload .env
        print(f"Uploading {local_env_path} to {remote_env_path}...")
        sftp = ssh.open_sftp()
        try:
            sftp.put(local_env_path, remote_env_path)
            print("Upload completed successfully!")
        finally:
            sftp.close()
            
        # 2. Execute git and docker-compose commands
        commands = [
            f"cd {remote_dir} && git fetch origin",
            f"cd {remote_dir} && git reset --hard origin/main",
            f"cd {remote_dir} && docker-compose up -d --build"
        ]
        
        for cmd in commands:
            print(f"\nExecuting: {cmd}")
            stdin, stdout, stderr = ssh.exec_command(cmd)
            
            # Read stdout and stderr in real-time or completely
            out = stdout.read().decode('utf-8', errors='ignore')
            err = stderr.read().decode('utf-8', errors='ignore')
            
            if out:
                print("[STDOUT]")
                print(out)
            if err:
                print("[STDERR]")
                print(err)
                
        print("\n=== DEPLOYMENT COMPLETED! ===")
        
    except Exception as e:
        print(f"An error occurred: {e}", file=sys.stderr)
        sys.exit(1)
    finally:
        ssh.close()

if __name__ == '__main__':
    main()

File: scratch/deploy_vps.js
const { execSync } = require('child_process');

console.log('=== INICIANDO FORCE DEPLOY NO SERVIDOR (192.168.1.70) ===');

try {
  console.log('1. Git pull origin main...');
  const pull = execSync('ssh -o StrictHostKeyChecking=no ed@192.168.1.70 "cd /home/ed/projetcs/BelaFarma && git pull origin main"', { encoding: 'utf-8' });
  console.log('Git Pull Output:\n', pull);

  console.log('2. Reconstruindo container frontend com --no-cache...');
  const rebuild = execSync('ssh -o StrictHostKeyChecking=no ed@192.168.1.70 "cd /home/ed/projetcs/BelaFarma && docker-compose build --no-cache frontend && docker-compose up -d --force-recreate frontend backend"', { encoding: 'utf-8' });
  console.log('Docker Output:\n', rebuild);

  console.log('=== FORCE DEPLOY CONCLUÍDO COM SUCESSO! ===');
} catch (err) {
  console.error('ERRO NO DEPLOY:', err.message);
  if (err.stdout) console.log('STDOUT:', err.stdout);
  if (err.stderr) console.log('STDERR:', err.stderr);
}
